src/DotLiquid.Tests/Tags/StandardTagTests.cs

```diff
@@ -701,5 +709,59 @@ public void TestLegacyKeyValueDrop()
             Assert.AreEqual("Green", drop.BeforeMethod("Last"));
             Assert.IsNull(drop.BeforeMethod("UnknownProperty"));
         }
+
+        [Test]
+        public void IncrementNonExistentValue()
+        {
+            var template = Template.Parse("{% increment counter %}{% increment counter %}{% increment counter %}");
+            var output = template.Render(CultureInfo.InvariantCulture);
+
+            Assert.AreEqual("012", output);
+        }
+
+        [Test]
+        public void IncrementDoesNotAlterLocalVariables()
+        {
+            var template = Template.Parse("{% increment counter %}-{{counter}}-{% increment counter %}-{{counter}}-{% increment counter %}");
+            var output = template.Render(new RenderParameters(CultureInfo.InvariantCulture)
+            {
+                LocalVariables = Hash.FromAnonymousObject(new { counter = 42 })
+            });
+
+            Assert.AreEqual("0-42-1-42-2", output);
+        }
+
+        [Test]
+        public void IncrementDetectsBadSyntax()
+        {
+            Assert.Throws<SyntaxException>(() => Template.Parse("{% increment %}"));

```

Comment:
Unfortunately the Ruby library works wonky on this. See https://replit.com/@microalps/LiquidIncrementTests#main.rb for unexpected outcome. I don't think the Ruby library is right, but to add a test that codifies our behavior as different than Ruby is also weird.

Reply:
Ok we should definitively not hide a behavior by not writing a test, especially if we're concerned about it. We can also have a test validation comment like "hey, this is the legacy behavior, be careful if you want to change it so not to break users".

I'm fine with either following the Ruby lead if the rest of our tags is doing it currently, or clarifying we're departing from it because we feel it is 'right', or if there is no precedence in other tags create an option (ugh) for users to choose their behavior (like we have for other 'Ruby' aspects).

Reply:
Added tests for increment/decrement and noted that the behavior is wonky and this test can be changed in the future.